src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -20,6 +20,8 @@ public static class ConfigurationBinder
         private const string InstanceGetTypeTrimmingWarningMessage = "Cannot statically analyze the type of instance so its members may be trimmed";
         private const string PropertyTrimmingWarningMessage = "Cannot statically analyze property.PropertyType so its members may be trimmed.";
         private const string BindSingleElementsToArraySwitch = "Microsoft.Extensions.Configuration.BindSingleElementsToArray";
+        private static bool TryGetSwitchBindSingleElementsToArrayFromAppContext = AppContext.TryGetSwitch(BindSingleElementsToArraySwitch, out BindSingleElementsToArray);
```

Comment:
This is a better pattern to use:

[https://github.com/dotnet/runtime/blob/3d5d26181cd7bc07ea6c6f87710c14ccd043b415/src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceProvider.cs#L33-L34](https://github.com/dotnet/runtime/blob/3d5d26181cd7bc07ea6c6f87710c14ccd043b415/src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceProvider.cs#L33-L34)